corehq/apps/app_manager/xform.py

```diff
@@ -495,32 +496,42 @@ def update_block(self):
         self.elem.append(update_block)
         return update_block
 
-    def add_case_updates(self, updates, make_relative=False):
+    def add_case_updates(self, updates, make_relative=False, domain=None):
+        from corehq.apps.app_manager.models import ConditionalCaseUpdate
         update_block = self.update_block
         if not updates:
             return
 
         update_mapping = {}
         attachments = {}
         for key, value in updates.items():
+            value_str = getattr(value, "question_path", value)
             if key == 'name':
                 key = 'case_name'
-            if self.is_attachment(value):
-                attachments[key] = value
+            if self.is_attachment(value_str):
+                attachments[key] = value_str
             else:
                 update_mapping[key] = value
 
         for key, q_path in sorted(update_mapping.items()):
+            resolved_path = self.xform.resolve_path(q_path)
+            edit_mode_path = ''
+            if (SAVE_ONLY_EDITED_FORM_FIELDS.enabled(domain, NAMESPACE_DOMAIN)
```

Comment:
@snopoke you're right the FF check should definitely be taken out of that for-loop. I can see either putting the check at the top of the `add_case_updates` method or putting it in the constructor for `XFormCaseBlock` (and just pass a domain to any new `XFormCaseBlock`). I would be against adding it to the constructor for `XForm` though, as that would extend the scope into several other files/parts of the codebase.

Reply:
`CommCareFeatureSupportMixin` sounds fine since this is a feature release toggle so the code won't last long. I'd rather not crowd that file with a lot of permanent toggle checks.